pinot-minion/src/main/java/org/apache/pinot/minion/executor/SegmentGenerationAndPushTaskExecutor.java

```diff
@@ -281,18 +282,27 @@ private SegmentGenerationTaskSpec generateTaskSpec(Map<String, String> taskConfi
     recordReaderSpec.setClassName(taskConfigs.get(BatchConfigProperties.RECORD_READER_CLASS));
     recordReaderSpec.setConfigClassName(taskConfigs.get(BatchConfigProperties.RECORD_READER_CONFIG_CLASS));
     taskSpec.setRecordReaderSpec(recordReaderSpec);
+
+    String rawTableName = taskConfigs.get(BatchConfigProperties.TABLE_NAME);
+    String tableNameWithType = rawTableName != null ? TableNameBuilder.OFFLINE.tableNameWithType(rawTableName) : null;
     Schema schema;
     if (taskConfigs.containsKey(BatchConfigProperties.SCHEMA)) {
       schema = JsonUtils
           .stringToObject(JsonUtils.objectToString(taskConfigs.get(BatchConfigProperties.SCHEMA)), Schema.class);
     } else if (taskConfigs.containsKey(BatchConfigProperties.SCHEMA_URI)) {
```

Comment:
Is this for backward-compatibility? I don't see where we set the uri in the task generator

Reply:
Yes. And I also want to keep a config to allow the override for schema and table config, basically the field value and URI.